### ISSUE TITLE
update issue - node-gyp error

### ISSUE THREAD

**Issue by ilpepe** [sharelatex/sharelatex#294]
Hi,
I was trying to update my sharelatex installation.

I used 
sudo git pull 
sudo npm install
sudo grunt install
sudo grunt update:all

The first two worked perfect, however, when running  "sudo grunt install" I got:
Severa of these warnings:

gyp ERR! clean error 
gyp ERR! stack Error: EACCES, unlink 'build'
gyp ERR! System Linux 3.16.0-4-amd64
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/sharelatex/spelling/node_modules/restify/node_modules/dtrace-provider
gyp ERR! node -v v0.10.29
gyp ERR! node-gyp -v v0.13.1
gyp ERR! not ok 

npm ERR! dtrace-provider@0.2.8 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the dtrace-provider@0.2.8 install script.
npm ERR! This is most likely a problem with the dtrace-provider package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls dtrace-provider
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.16.0-4-amd64
npm ERR! command "/usr/local/bin/node" "/usr/local/bin/npm" "rebuild"
npm ERR! cwd /opt/sharelatex/spelling
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.14
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /opt/sharelatex/spelling/npm-debug.log
npm ERR! not ok code 0

A bit further in the ouput I got done, with errors, so I continued.
The site is online and I can see all the projects. I can open then, but hen I got an error message. I can't see the tex file nor the pdf, but I can see what files are available within the project.

Clues for fixing?
The app runs on debian.

regards,
Pieter

**Comment by ilpepe** [comment]
update:
output running sharelatex:
[2015-04-03T13:30:54.542Z] ERROR: docstore/21145 on ibvs09: request errored
    NotFoundError: No such doc: 54ae94091e1cb77f399b3a92 in project 54abf6331e1cb77f399b3a55
        at new NotFoundError (/opt/sharelatex/docstore/app/js/Errors.js:6:13)
        at /opt/sharelatex/docstore/app/js/DocManager.js:23:27
        at /opt/sharelatex/docstore/app/js/MongoManager.js:17:16
        at /opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/cursor.js:158:16
        at commandHandler (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/cursor.js:651:16)
        at /opt/sharelatex/docstore/node_modules/metrics-sharelatex/mongodb.coffee:42:29
        at /opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/db.js:1670:9
        at Server.Base._callHandler (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/connection/base.js:382:41)
        at /opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/connection/server.js:472:18
        at MongoReply.parseBody (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
        at null.<anonymous> (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/connection/server.js:430:20)
        at emit (events.js:95:17)
        at null.<anonymous> (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:191:13)
        at emit (events.js:98:17)
        at Socket.<anonymous> (/opt/sharelatex/docstore/node_modules/mongojs/node_modules/mongodb/lib/mongodb/connection/connection.js:384:22)
        at Socket.emit (events.js:95:17)
        at Socket.<anonymous> (_stream_readable.js:748:14)
        at Socket.emit (events.js:92:17)
        at emitReadable_ (_stream_readable.js:410:10)
        at emitReadable (_stream_readable.js:406:5)
        at readableAddChunk (_stream_readable.js:168:9)
        at Socket.Readable.push (_stream_readable.js:130:10)
        at TCP.onread (net.js:528:21)
[2015-04-03T13:30:54.546Z] ERROR: web-sharelatex/21120 on ibvs09: error getting doc from docstore (project_id=54abf6331e1cb77f399b3a55, doc_id=54ae94091e1cb77f399b3a92)
    Error: docstore api responded with non-success code: 404
      at Request._callback (/opt/sharelatex/web/app/js/Features/Docstore/DocstoreManager.js:108:19)
      at Request.self.callback (/opt/sharelatex/web/node_modules/request/request.js:121:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:978:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:929:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:929:16
      at process._tickDomainCallback (node.js:463:13)
    
[2015-04-03T13:30:54.546Z]  INFO: documentupdater/21136 on ibvs09: flushing docs (project_id=54abf6331e1cb77f399b3a55, doc_ids=[])
[2015-04-03T13:30:54.546Z] ERROR: web-sharelatex/21120 on ibvs09: error finding element for getDocument (doc_id=54ae94091e1cb77f399b3a92, project_id=54abf6331e1cb77f399b3a55)
    Error: docstore api responded with non-success code: 404
      at Request._callback (/opt/sharelatex/web/app/js/Features/Docstore/DocstoreManager.js:108:19)
      at Request.self.callback (/opt/sharelatex/web/node_modules/request/request.js:121:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:978:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:929:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:929:16
      at process._tickDomainCallback (node.js:463:13)
    
[2015-04-03T13:30:54.546Z]  INFO: documentupdater/21136 on ibvs09: flushed project via http (project_id=54abf6331e1cb77f399b3a55)
[2015-04-03T13:30:54.544Z]  INFO: docstore/21145 on ibvs09: http request (response-time=4, req.remote-addr=127.0.0.1)
    GET /project/54abf6331e1cb77f399b3a55/doc/54ae94091e1cb77f399b3a92 HTTP/1.1
[2015-04-03T13:30:54.547Z] ERROR: web-sharelatex/21120 on ibvs09: error passed to top level next middlewear (method=GET)
    Error: docstore api responded with non-success code: 404
      at Request._callback (/opt/sharelatex/web/app/js/Features/Docstore/DocstoreManager.js:108:19)
      at Request.self.callback (/opt/sharelatex/web/node_modules/request/request.js:121:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:978:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/opt/sharelatex/web/node_modules/request/request.js:929:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:929:16
      at process._tickDomainCallback (node.js:463:13)
    
    --
    url: /project/54abf6331e1cb77f399b3a55/doc/54ae94091e1cb77f399b3a92

and  so on.

So, how I see it, somehow a wrong install of docstore, and therfor, it cannot serve the file tot the app.
That is probably why the app is working except for showing and compiling the files...

**Comment by ilpepe** [comment]
Hi,

I now logged in as root, instead of loggin in as user and then going to root, and it worked.

However it seems that I can see a single file, and I'm not able to compile when using multiple files...

latexmk errors spilled out:
[2015-04-03T14:45:16.931Z]  WARN: clsi/32749 on ibvs09: timeouts and sandboxing are not enabled with CommandRunner
Latexmk: This is Latexmk, John Collins, 10 Nov 2013, version: 4.39.
*** Report bugs etc to John Collins <collins at phys.psu.edu>. ***
Latexmk: Changing directory to '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/'
Rule 'pdflatex': File changes, etc:
   Changed files, or newly in use since previous run(s):
      '1_main.tex'
   Non-existent destination files:
      '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/output.pdf'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex -synctex=1 -interaction=batchmode  -recorder -output-directory="/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55" --jobname=output  "1_main.tex"'
------------
Latexmk: applying rule 'pdflatex'...
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2015/dev/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
'output.fls' and '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/output.fls' are identical (not copied) at /usr/local/bin/latexmk line 6454.
Latexmk: Index file 'output.idx' was written
Latexmk: Missing input file: 'Abbreviations.tex' from line
  'No file Abbreviations.tex.'
Latexmk: Missing input file: 'OverviewToDos.tex' from line
  'No file OverviewToDos.tex.'
Latexmk: Missing input file: 'MindMap.tex' from line
  'No file MindMap.tex.'
Latexmk: Missing input file: 'Introduction.tex' from line
  '! LaTeX Error: File `Introduction.tex' not found.'
Rule 'pdflatex': File changes, etc:
   Changed files, or newly in use since previous run(s):
      '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/output.aux'
      'output.aux'
------------
Run number 2 of rule 'pdflatex'
------------
------------
Running 'pdflatex -synctex=1 -interaction=batchmode  -recorder -output-directory="/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55" --jobname=output  "1_main.tex"'
------------
Latexmk: applying rule 'pdflatex'...
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2015/dev/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
'output.fls' and '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/output.fls' are identical (not copied) at /usr/local/bin/latexmk line 6454.
Latexmk: Index file 'output.idx' was written
Latexmk: Missing input file: 'Abbreviations.tex' from line
  'No file Abbreviations.tex.'
Latexmk: Missing input file: 'OverviewToDos.tex' from line
  'No file OverviewToDos.tex.'
Latexmk: Missing input file: 'MindMap.tex' from line
  'No file MindMap.tex.'
Latexmk: Missing input file: 'Introduction.tex' from line
  '! LaTeX Error: File `Introduction.tex' not found.'
Failure to make '/opt/sharelatex/data/compiles/54abf6331e1cb77f399b3a55/output.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 256
Latexmk: Undoing directory change

It seems the necessary files can't be found, however I can select them in sharelatex without opening them.

Needs some help
